crucible-mir/src/Mir/TransCustom.hs

```diff
@@ -761,6 +763,21 @@ intrinsics_copy_nonoverlapping = ( ["core", "intrinsics", "copy_nonoverlapping"]
             _ -> mirFail $ "bad arguments for intrinsics::copy_nonoverlapping: " ++ show ops
         _ -> Nothing)
 
+cell_swap_is_nonoverlapping :: (ExplodedDefId, CustomRHS)
+cell_swap_is_nonoverlapping =
+    ( ["core", "cell", "{impl}", "swap", "crucible_cell_swap_is_nonoverlapping_hook"]
+    , \case
+        Substs [ty] -> Just $ CustomOp $ \_ ops -> case ops of
+            [MirExp MirReferenceRepr src,
+             MirExp MirReferenceRepr dest] -> do
+                size <- getLayoutFieldAsExpr "crucible_cell_swap_is_nonoverlapping_hook" laySize ty
+                MirExp C.BoolRepr <$> isNonOverlapping src dest size
+            _ -> mirFail $
+                "bad arguments for Cell::swap::crucible_cell_swap_is_nonoverlapping_hook: "
+                ++ show ops
+        _ -> Nothing
+    )
```

Comment:
Is there a `crux-mir` test case that uses `Cell::swap` in order to ensure that this override has test coverage?

Reply:
Yeah, it's in `test/conc_eval/cell/swap.rs`. There are also a few `crux-mir-comp` tests that use it.